Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/InteractableToggleCollection.cs

```diff
@@ -11,78 +12,108 @@ namespace Microsoft.MixedReality.Toolkit.UI
-                ToggleList[i].CanDeselect = false;
+                return currentIndex;
+            }
+            set
+            {
+                SetSelection(value, false, true);
             }
-
-            OnSelection(CurrentIndex, true);
         }
 
-        private void OnEnable()
+        [Tooltip("This event is triggered when any of the toggles in the ToggleList are selected")]
+        /// <summary>
+        /// This event is triggered when any of the toggles in the ToggleList are selected
+        /// </summary>
+        public UnityEvent OnSelectionEvents = new UnityEvent();
+
+        private void Start()
         {
-            OnSelection(CurrentIndex, true);
+            if (ToggleList != null)
+            {
+                // Add listeners to each toggle in ToggleList
+                for (int i = 0; i < ToggleList.Length; ++i)
+                {
+                    int itemIndex = i;
+                    ToggleList[i].OnClick.AddListener(() => SetSelection(itemIndex, true, false));
+                    ToggleList[i].CanDeselect = false;
+                }
+
+                SetSelection(CurrentIndex, true, true);
+            }
         }
 
         /// <summary>
-        /// Sets the selected index and selected Interactive
+        /// Set the selection of a an element in the toggle collection based on index.
+        /// <param name="index">Index of an element in ToggleList</param>
+        /// <param name="force">Force update currentIndex</param>
+        /// <param name="fireOnClick">The manual trigger of the OnClick event. OnClick event is manually triggered 
+        /// when the CurrentIndex is updated via script or inspector</param>
         /// </summary>
-        public void SetSelection(int index)
+        public void SetSelection(int index, bool force = false, bool fireOnClick = false)
         {
-            if (!isActiveAndEnabled ||
-                (index < 0 || ToggleList.Length <= index))
+            if (index < 0 || ToggleList.Length <= index)
             {
+                Debug.LogWarning("Index out of range of ToggleList: " + index);
                 return;
             }
 
-            OnSelection(index, true);
+            if (CurrentIndex != index || force)
+            {
+                currentIndex = index;
+
+                OnSelection(index);
+            }
+
+            if (fireOnClick)
+            {
+                ToggleList[index].TriggerOnClick();
+            }
         }
 
-        /// <summary>
-        /// Set the toggle state of each button based on the selected item
-        /// </summary>
+        // Update the visual appearance and set the states of the selected and unselected toggles within 
+        // Interactable
         protected virtual void OnSelection(int index, bool force = false)
         {
-            for (int i = 0; i < ToggleList.Length; ++i)
+            int length = ToggleList.Length;
```

Comment:
super minor, but you don't need to pull this out into a separate variable lengths, since length is just used once. Just stylistically to be consistent with other code.

Reply:
Accidentally added the wrong reaction, meant to do thumbs up